pkg/codegen/nodejs/gen.go

```diff
@@ -795,9 +795,14 @@ func (mod *modContext) genResource(w io.Writer, r *schema.Resource) (resourceFil
 					arg = fmt.Sprintf("(%s) ?? %s", arg, dv)
 				}
 
-				// provider properties must be marshaled as JSON strings.
 				if r.IsProvider && !isStringType(prop.Type) {
-					arg = fmt.Sprintf("pulumi.output(%s).apply(JSON.stringify)", arg)
+					encode := !mod.getNodePackageInfo().DoNotJSONEncodeProviderConfiguration
```

Comment:
Subtlety: should the PackageInfo be looked up from `mod` (mod.pkg) or should it be looked up from package hosting `r`? Is this code ever exercised for foreign properties?